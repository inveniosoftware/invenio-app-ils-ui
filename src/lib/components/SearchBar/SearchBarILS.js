import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Input } from 'semantic-ui-react';
import { screenIsWiderThan } from '@components/utils';
import { Breakpoints } from '@components/Media';

export class SearchBarILS extends Component {
  state = { currentValue: '' };

  componentDidMount() {
    if (this.focusInput) {
      this.focusInput.focus();
    }
  }

  clearQueryString = () => {
    this.setState({ currentValue: '' });
  };

  onKeyPressHandler = (event, input) => {
    if (event.key === 'Enter') {
      const { onSearchHandler } = this.props;
      const { currentValue } = this.state;
      onSearchHandler(currentValue);
    }
  };

  onPasteHandler = (event) => {
    const { onSearchHandler } = this.props;
    event.preventDefault();
    const queryString = (event.clipboardData || window.clipboardData).getData(
      'text'
    );
    this.setState({ currentValue: queryString });
    onSearchHandler(queryString);
  };

  render() {
    const {
      className: parentClass,
      onKeyPressHandler: parentKeyPressHandler,
      onSearchHandler,
      onPasteHandler,
      onChangeHandler,
      placeholder,
      responsiveAutofocus,
      ...rest
    } = this.props;
    const { currentValue } = this.state;
    const autofocus = screenIsWiderThan(Breakpoints.computer);
    return (
      <Input
        action={{
          icon: 'search',
          onClick: () => onSearchHandler(currentValue),
        }}
        onChange={(event, { value }) => {
          this.setState({ currentValue: value });
          onChangeHandler && onChangeHandler(value);
        }}
        value={currentValue}
        onKeyPress={parentKeyPressHandler || this.onKeyPressHandler}
        onPaste={onPasteHandler || this.onPasteHandler}
        fluid
        size="big"
        placeholder={placeholder}
        className={`${parentClass} ils-searchbar`}
        ref={(input) => {
          if (responsiveAutofocus && autofocus) {
            this.focusInput = input;
          } else if (!responsiveAutofocus) {
            this.focusInput = input;
          }
        }}
        {...rest}
      />
    );
  }
}

SearchBarILS.propTypes = {
  onKeyPressHandler: PropTypes.func,
  ref: PropTypes.func,
  onPasteHandler: PropTypes.func,
  onSearchHandler: PropTypes.func.isRequired,
  onChangeHandler: PropTypes.func,
  placeholder: PropTypes.string,
  className: PropTypes.string,
  responsiveAutofocus: PropTypes.bool,
};

SearchBarILS.defaultProps = {
  onKeyPressHandler: null,
  onPasteHandler: null,
  onChangeHandler: null,
  placeholder: '',
  className: '',
  responsiveAutofocus: false,
  ref: null,
};
