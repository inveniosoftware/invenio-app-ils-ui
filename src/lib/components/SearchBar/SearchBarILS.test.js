import React from 'react';
import { mount } from 'enzyme';
import { SearchBarILS as SearchBar } from './SearchBarILS';

const mockedOnSearchHandler = jest.fn();

describe('SearchBar tests', () => {
  let component;
  afterEach(() => {
    mockedOnSearchHandler.mockClear();
    component.unmount();
  });

  it('should match snapshot', () => {
    component = mount(
      <SearchBar onSearchHandler={() => {}} placeholder="Search" />
    );
    expect(component).toMatchSnapshot();
  });

  it('should call onSearchHandler only on `enter` key press', () => {
    component = mount(
      <SearchBar placeholder="Search" onSearchHandler={mockedOnSearchHandler} />
    );

    const input = component.find('Input').find('input');
    input.simulate('keypress', { key: 'Enter' });
    expect(mockedOnSearchHandler).toHaveBeenCalled();

    mockedOnSearchHandler.mockClear();

    input.simulate('keypress', { key: 'Backspace' });
    expect(mockedOnSearchHandler).not.toHaveBeenCalled();
  });

  it('should call executeSearch on button click', () => {
    component = mount(
      <SearchBar onSearchHandler={mockedOnSearchHandler} placeholder="Search" />
    );
    const input = component.find('Input').find('Icon').find('i');
    input.simulate('click');
    expect(mockedOnSearchHandler).toHaveBeenCalled();
  });
});
