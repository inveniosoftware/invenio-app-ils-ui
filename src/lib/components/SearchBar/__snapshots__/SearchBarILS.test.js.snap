// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar tests should match snapshot 1`] = `
<SearchBarILS
  className=""
  onChangeHandler={null}
  onKeyPressHandler={null}
  onPasteHandler={null}
  onSearchHandler={[Function]}
  placeholder="Search"
  responsiveAutofocus={false}
>
  <Input
    action={
      Object {
        "icon": "search",
        "onClick": [Function],
      }
    }
    className=" ils-searchbar"
    fluid={true}
    onChange={[Function]}
    onKeyPress={[Function]}
    onPaste={[Function]}
    placeholder="Search"
    size="big"
    type="text"
  >
    <div
      className="ui big fluid action input  ils-searchbar"
      onPaste={[Function]}
    >
      <input
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search"
        type="text"
      />
      <Button
        as="button"
        icon="search"
        onClick={[Function]}
      >
        <Ref
          innerRef={
            Object {
              "current": <button
                class="ui icon button"
              >
                <i
                  aria-hidden="true"
                  class="search icon"
                />
              </button>,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <button
                  class="ui icon button"
                >
                  <i
                    aria-hidden="true"
                    class="search icon"
                  />
                </button>,
              }
            }
          >
            <button
              className="ui icon button"
              onClick={[Function]}
            >
              <Icon
                as="i"
                name="search"
              >
                <i
                  aria-hidden="true"
                  className="search icon"
                  onClick={[Function]}
                />
              </Icon>
            </button>
          </RefFindNode>
        </Ref>
      </Button>
    </div>
  </Input>
</SearchBarILS>
`;
